### ISSUE TITLE
Support password credentials flow for federated users

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/azure-sdk-for-go#17615]
Ported from https://github.com/Azure/go-autorest/issues/378

**Comment by chlowell** [comment]
Closing this as a duplicate of #14405. Current status is that we're waiting on MSAL to support federated user authentication.
